### ISSUE TITLE
Keybase broke my network

### ISSUE THREAD

**Issue by smitelli** [keybase/client#6690]
I'll preface this by saying that, based on my searching, I'm the only one this happened to. If this was a widespread issue I'm sure I wouldn't have been the first person that encountered it. This issue may not reproduce, but I'm posting it here in case it helps somebody who sees the same problem.

OS is Windows 7 SP1 x64. Keybase client version is whatever was [here](https://keybase.io/docs/the_app/install_windows) on April 20.

I installed the Keybase client app and immediately noticed that I could no longer connect to any network shares through standard methods.

* The Network folder and all network discovery areas listed no other devices other than the local machine.
* In Windows Explorer, typing `\\server\share` in the address bar resulted in the error "Windows cannot access '\\server\share'" along with the code "0x80004005, Unspecified error."
* From the command prompt, `net use z: \\server\share` resulted in the error "The Workstation service has not been started."
* One of the only successful methods of accessing the share was through the command prompt: `dir \\server\share` worked properly, as did other command-line methods of interacting with files.

I tried many fruitless and disruptive fixes, none of which worked. I was almost ready to give up and reinstall Windows when I stumbled on a Super User post that led me on the right path:

In my registry, the key `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\NetworkProvider\Order` had the string `ProviderOrder` set to `,Dokan1` -- literally a comma followed by Dokan1 -- and similar values were found in `ControlSet001`, `002`, etc. Dokan appears to be something that the Keybase virtual filesystem installed, which is something that the Keybase client installed, which is why I'm opening this here.

Looks to me like the installer tried to append Dokan to the existing list like

    ProviderOrder = ProviderOrder + ",Dokan1"

but whatever was present on my system ended up clobbering the old value instead of appending to it.

I don't know what that registry value was set to before. System Restore was not useful in trying to recover the old value, nor is this something I have in any of my personal backups. The best I could do was copy the `ProviderOrder` value from a known-working system, `RDPNP,LanmanWorkstation,webclient`. That resolved the issue.

I have since uninstalled Dokan, and the Keybase client.

**Comment by cjb** [comment]
Thanks for the report, and apologies for the trouble.  We'll try to reproduce and check it out.

CC @zanderz @taruti

**Comment by zanderz** [comment]
You are right, the Dokan mounting utility seems to write the comma regardless of what was there:
https://github.com/dokan-dev/dokany/blob/b4dc99ad9efa0b60b05a1e37653e1fc1ba793f43/dokan/mount.c#L315
and also seems a bit cavalier about checking errors from reading the existing string. However, I am pretty sure it was not supposed to be blank in the first place - mine also has `RDPNP,LanmanWorkstation,webclient`.

We frequently sanity check installations on Win7 x64, so that shouldn't be a problem by itself. Do you have antivirus software installed, or something that may monitor and block registry access? Are you willing to try installing again, maybe with a system restore point ready just in case? Lastly, you can invoke our installer exe with the argument `nodriver=1` to skip the Dokan part, so you can still use Keybase chat and `keybase fs` commands.

**Comment by taruti** [comment]
Also we don't need the Dokan network provider, so just avoiding that registry write could be a possible solution?
